src/hmac.rs

```diff
@@ -364,4 +396,40 @@ mod tests {
             assert!(hmac::verify(&key, HELLO_WORLD_BAD, tag.as_ref()).is_err())
         }
     }
+
+    #[test]
+    pub fn hmac_generate_with_length() {
```

Comment:
Tests of the public HMAC API should go in `hmac_tests.rs`.

I will write a test for the original `generate()` that you can emulate.